apps/laboratory/tests/verify.spec.ts

```diff
@@ -0,0 +1,120 @@
+
+    await modalPage.disconnect()
+    await modalValidator.expectDisconnected()
+    await walletValidator.expectDisconnected()
+  }
+)
+
+testMVerifyValid(
+  'connection and signature requests from non-scam verified domain should show as domain match',
+  async ({ modalPage, modalValidator, context }) => {
+    const walletPage = new WalletPage(await context.newPage())
+    await walletPage.load()
+    const walletValidator = new WalletValidator(walletPage.page)
+
+    const uri = await modalPage.getConnectUri()
+    await walletPage.connectWithUri(uri)
+    await expect(walletPage.page.getByTestId('session-info-verified')).toBeVisible()
+    await walletPage.handleSessionProposal(DEFAULT_SESSION_PARAMS)
+    await modalValidator.expectConnected()
+    await walletValidator.expectConnected()
+
+    await modalPage.sign()
+    const chainName = modalPage.library === 'solana' ? 'Solana' : DEFAULT_CHAIN_NAME
+    await walletValidator.expectReceivedSign({ chainName })
+    await expect(walletPage.page.getByTestId('session-info-verified')).toBeVisible()
+    await walletPage.handleRequest({ accept: true })
+    await modalValidator.expectAcceptedSign()
+
+    await modalPage.disconnect()
+    await modalValidator.expectDisconnected()
+    await walletValidator.expectDisconnected()
+  }
+)
+
+testMVerifyDomainMismatch(
+  'connection and signature requests from non-scam verified domain but on localhost should show as invalid domain',
+  async ({ modalPage, modalValidator, context }) => {
+    const walletPage = new WalletPage(await context.newPage())
+    await walletPage.load()
+    const walletValidator = new WalletValidator(walletPage.page)
+
+    const uri = await modalPage.getConnectUri()
+    await walletPage.connectWithUri(uri)
+    await expect(walletPage.page.getByText('Invalid Domain')).toBeVisible()
+    await walletPage.handleSessionProposal(DEFAULT_SESSION_PARAMS)
+    await modalValidator.expectConnected()
+    await walletValidator.expectConnected()
+
+    await modalPage.sign()
+    const chainName = modalPage.library === 'solana' ? 'Solana' : DEFAULT_CHAIN_NAME
+    await walletValidator.expectReceivedSign({ chainName })
+    await expect(walletPage.page.getByText('Invalid Domain')).toBeVisible()
+    await walletPage.handleRequest({ accept: true })
+    await modalValidator.expectAcceptedSign()
+
+    await modalPage.disconnect()
+    await modalValidator.expectDisconnected()
+    await walletValidator.expectDisconnected()
+  }
+)
+
+testMVerifyEvil(
+  'connection and signature requests from scam verified domain should show as scam domain',
+  async ({ modalPage, modalValidator, context }) => {
+    const walletPage = new WalletPage(await context.newPage())
+    await walletPage.load()
+    const walletValidator = new WalletValidator(walletPage.page)
+
+    const uri = await modalPage.getConnectUri()
+    await walletPage.connectWithUri(uri)
+    await expect(walletPage.page.getByText('Website flagged')).toBeVisible()
```

Comment:
shouldn't this be on the validator?

Reply:
What do you mean?

Reply:
assertions are generally on the validator (ModalValidator)

Reply:
I don't think we need to over-abstract this. IMO fine as-is